### PR TITLE
Rfr 80 - Remake Login Page in MVP

### PR DESCRIPTION
**Pull request by fin1989**

### CODE REVIEW

File: app/src/main/java/rss/feed/reader/presenters/LoginPresenter.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+public class LoginPresenter implements LoginContract.Presenter {
+
+    public UserManager mUserManager;
+
+    private static final String emailPattern = "(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])";
+
+    private LoginContract.View mView;
+    private Context mContext;
+
+    public LoginPresenter(LoginContract.View view, Context context, UserManager userManager){
```
Comment (qUser13) [inline app/src/main/java/rss/feed/reader/presenters/LoginPresenter.java]: How do you feel about making presenter a 'pure' with no dependencies on Android framework?

File: app/src/main/java/rss/feed/reader/contracts/LoginContract.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        void hideProgress();
+
+        void showSnackBar(String message);
+
+        void navigateToMain();
+    }
+
+    interface Presenter {
+
+        void signIn(EditText email, EditText password);
```
Comment (qUser13) [inline app/src/main/java/rss/feed/reader/contracts/LoginContract.java]: Preferable, presenter should have no reference on view components directly.

File: app/src/main/java/rss/feed/reader/presenters/LoginPresenter.java (hunk 1 above)
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/presenters/LoginPresenter.java]: +1

File: app/src/main/java/rss/feed/reader/contracts/LoginContract.java (hunk 2 above)
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/contracts/LoginContract.java]: +1

File: app/src/main/java/rss/feed/reader/presenters/LoginPresenter.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import rss.feed.reader.managers.base.UserManager;
+import rss.feed.reader.utils.listeners.OnActionComplete;
+
+/**
+ * Created by omartynets on 12.02.2017.
+ */
+
+public class LoginPresenter implements LoginContract.Presenter {
+
+    public UserManager mUserManager;
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/presenters/LoginPresenter.java]: I guess it's better to use package-level visibility,  without "public" access level

File: app/src/main/java/rss/feed/reader/presenters/LoginPresenter.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+                } else {
+                    mView.showSnackBar(reason);
+                }
+            }
+        });
+        }
+    }
+
+    private boolean isValidCredentials(EditText emailField, @Nullable EditText passwordField) {
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/presenters/LoginPresenter.java]: can passwordField be null ?

File: app/src/main/java/rss/feed/reader/ui/activities/SignInActivity.java [hunk 5]
```diff
@@ -5,19 +5,28 @@
...
-import rss.feed.reader.utils.listeners.OnActionComplete;
 
 /**
  * Created by omartynets on 27.10.2016.
  */
-public class SignInActivity extends BaseSignActivity {
+public class SignInActivity extends BaseActivity implements LoginContract.View {
+
+    @Inject
+    public UserManager mUserManager;
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/ui/activities/SignInActivity.java]: we could remove "public" I guess

File: app/src/main/java/rss/feed/reader/presenters/LoginPresenter.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        mView = view;
+        this.mContext = context;
+        mUserManager = userManager;
+    }
+
+    @Override
+    public void signIn(EditText email, EditText password) {
+
+        if(isValidCredentials(email, password)) {
+        mView.showProgress(R.string.progress_message_authenticating);
```
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/presenters/LoginPresenter.java]: I would add extra spaces here, for the "if" block

**Comment by Vovchyk** [comment]
>> How about some tests?
yeah, it would be nice :)

File: app/src/main/java/rss/feed/reader/ui/activities/SignInActivity.java (hunk 5 above)
Comment (Vovchyk) [inline app/src/main/java/rss/feed/reader/ui/activities/SignInActivity.java]: any reason `mUserManager ` should be public ? ;)

**Pull request merged**